src/Lecture3.hs

```diff
@@ -49,15 +48,24 @@ data Weekday
     | Friday
     | Saturday
     | Sunday
-    deriving (Show, Eq)
+    deriving (Show, Eq, Enum, Bounded)
 
 {- | Write a function that will display only the first three letters
 of a weekday.
 
 >>> toShortString Monday
 "Mon"
+
 -}
-toShortString = error "TODO"
+
+toShortString :: Weekday -> String
+toShortString Monday = "Mon"
+toShortString Tuesday = "Tue"
+toShortString Wednesday = "Wed"
+toShortString Thursday = "Thu"
+toShortString Friday = "Fri"
+toShortString Saturday = "Sat"
+toShortString Sunday = "Sun"
```

Comment:
That's the correct implementation 👍🏻 
But can you implement it without pattern matching on each constructor? 🤔

Reply:
Nice. As `Weekday` has derived `Show`, I could use another way.